tests/orchestrator/test_cli_adapters_live.py

```
@@ -0,0 +1,517 @@
+
+
+def test_codex_entrypoint_selects_custom_provider():
+    """Run the production entrypoint with OPENAI_BASE_URL set, then assert
+    the rendered config has top-level `model_provider = "custom"` so the
+    custom block is actually used (not dead config)."""
+    if not _have("docker"):
+        pytest.skip("docker not installed")
+    if not _image_exists(WORKSPACE_IMAGE):
+        pytest.skip(f"workspace image '{WORKSPACE_IMAGE}' not present locally")
+
+    res = subprocess.run(
```

Ruff (0.15.11)

[error] `subprocess` call: check for execution of untrusted input

(S603)